applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -443,32 +494,61 @@ int streamctrl_start(void)
 	ret = data_fifo_init(&ble_fifo_rx);
 	ERR_CHK_MSG(ret, "Failed to set up ble_rx FIFO");
 
-	button_msg_sub_thread_id =
-		k_thread_create(&button_msg_sub_thread_data, button_msg_sub_thread_stack,
-				CONFIG_BUTTON_MSG_SUB_STACK_SIZE,
-				(k_thread_entry_t)button_msg_sub_thread, NULL, NULL, NULL,
-				K_PRIO_PREEMPT(CONFIG_BUTTON_MSG_SUB_THREAD_PRIO), 0, K_NO_WAIT);
+	button_msg_sub_thread_id = k_thread_create(
+		&button_msg_sub_thread_data, button_msg_sub_thread_stack,
+		CONFIG_BUTTON_MSG_SUB_STACK_SIZE, (k_thread_entry_t)button_msg_sub_thread, NULL,
+		NULL, NULL, K_PRIO_PREEMPT(CONFIG_BUTTON_MSG_SUB_THREAD_PRIO), 0, K_NO_WAIT);
 	ret = k_thread_name_set(button_msg_sub_thread_id, "BUTTON_MSG_SUB");
 	ERR_CHK(ret);
 
-	le_audio_msg_sub_thread_id =
-		k_thread_create(&le_audio_msg_sub_thread_data, le_audio_msg_sub_thread_stack,
-				CONFIG_LE_AUDIO_MSG_SUB_STACK_SIZE,
-				(k_thread_entry_t)le_audio_msg_sub_thread, NULL, NULL, NULL,
-				K_PRIO_PREEMPT(CONFIG_LE_AUDIO_MSG_SUB_THREAD_PRIO), 0, K_NO_WAIT);
+	le_audio_msg_sub_thread_id = k_thread_create(
+		&le_audio_msg_sub_thread_data, le_audio_msg_sub_thread_stack,
+		CONFIG_LE_AUDIO_MSG_SUB_STACK_SIZE, (k_thread_entry_t)le_audio_msg_sub_thread, NULL,
+		NULL, NULL, K_PRIO_PREEMPT(CONFIG_LE_AUDIO_MSG_SUB_THREAD_PRIO), 0, K_NO_WAIT);
 	ret = k_thread_name_set(le_audio_msg_sub_thread_id, "LE_AUDIO_MSG_SUB");
 	ERR_CHK(ret);
 
-	audio_datapath_thread_id =
-		k_thread_create(&audio_datapath_thread_data, audio_datapath_thread_stack,
-				CONFIG_AUDIO_DATAPATH_STACK_SIZE,
-				(k_thread_entry_t)audio_datapath_thread, NULL, NULL, NULL,
-				K_PRIO_PREEMPT(CONFIG_AUDIO_DATAPATH_THREAD_PRIO), 0, K_NO_WAIT);
+	audio_datapath_thread_id = k_thread_create(
+		&audio_datapath_thread_data, audio_datapath_thread_stack,
+		CONFIG_AUDIO_DATAPATH_STACK_SIZE, (k_thread_entry_t)audio_datapath_thread, NULL,
+		NULL, NULL, K_PRIO_PREEMPT(CONFIG_AUDIO_DATAPATH_THREAD_PRIO), 0, K_NO_WAIT);
 	ret = k_thread_name_set(audio_datapath_thread_id, "AUDIO DATAPATH");
 	ERR_CHK(ret);
 
-	ret = le_audio_enable(le_audio_rx_data_handler, audio_datapath_sdu_ref_update);
+	ret = le_audio_enable(le_audio_rx_data_handler, audio_datapath_sdu_ref_update,
+			      nonvalid_iso_cfgs);
 	ERR_CHK_MSG(ret, "Failed to enable LE Audio");
 
+	ret = bt_mgmt_init();
+	ERR_CHK(ret);
+
+	if ((CONFIG_AUDIO_DEV == HEADSET) && IS_ENABLED(CONFIG_TRANSPORT_CIS)) {
+		size_t ext_adv_size = 0;
+		const struct bt_data *ext_adv = NULL;
+
+		le_audio_adv_get(&ext_adv, &ext_adv_size, false);
+
+		ret = bt_mgmt_adv_start(ext_adv, ext_adv_size, NULL, 0, true);
+		ERR_CHK(ret);
+	} else if ((CONFIG_AUDIO_DEV == GATEWAY) && IS_ENABLED(CONFIG_TRANSPORT_BIS)) {
+		size_t ext_adv_size = 0;
+		size_t per_adv_size = 0;
+		const struct bt_data *ext_adv = NULL;
+		const struct bt_data *per_adv = NULL;
+
+		le_audio_adv_get(&ext_adv, &ext_adv_size, false);
```

Comment:
These names, can we make them more self-explanatory?

Reply:
These are used by functions like bt_le_ext_adv_set_data and bt_le_ext_adv_update_param, so I think the names should reflect that. Did you want to change them to extended_advertising_data? I think that might be too long